scimath/units/unit_array.py

```diff
@@ -68,17 +68,13 @@ class UnitArray(numpy.ndarray):
     # object interface
     ############################################################################
     def __repr__(self):
-        # this is a little more complicated than it should be in order
-        # to be more resilient to changes to numpy ndarray API
-        base_str = numpy.ndarray.__repr__(self.view(numpy.ndarray))
-        start = base_str.find('(')
-        end = base_str.rfind(')')
-
-        if start > -1 and end > -1:
-            base_str = base_str[start+1:end]
-
+        base_str = self._get_values_base_str()
         return "UnitArray(%s, units='%s')" % (base_str, repr(self.units))
 
+    def __str__(self):
+        base_str = self._get_values_base_str()
+        return "UnitArray(%s, units='%s')" % (base_str, self.units.label)
```

Comment:
I'm partial to `str.format` myself, but I won't insist... I do prefer to see the use of `self.__class__.__name__`, which behaves better under subclassing:
```
return "{}({}, units={!r})".format(self.__class__.__name__, base_str, self.units.label)
```